src/uu/ls/src/ls.rs

```diff
@@ -1821,78 +1824,67 @@ impl PathData {
             Dereference::None => false,
         };
 
-        let de: Option<DirEntry> = match dir_entry {
-            Some(de) => de.ok(),
-            None => None,
-        };
-
         // Why prefer to check the DirEntry file_type()?  B/c the call is
         // nearly free compared to a metadata() call on a Path
-        fn get_file_type(
-            de: &DirEntry,
-            p_buf: &Path,
-            must_dereference: bool,
-        ) -> OnceCell<Option<FileType>> {
+        let ft: OnceCell<Option<FileType>> = OnceCell::new();
+        let md: OnceCell<Option<Metadata>> = OnceCell::new();
+        let security_context: OnceCell<Box<str>> = OnceCell::new();
+
+        let de: RefCell<Option<Box<DirEntry>>> = if let Some(de) = dir_entry {
```

Comment:
did you try RefCell<Option<DirEntry>> ?

Reply:
My thinking was -- and is most likely silly -- this reduces the size of the struct.  Yes, it is more indirection and complexity.  Will try without.